examples/test_util/EclFile.cpp

```diff
@@ -16,490 +16,835 @@
+                                          };
+    return readBinaryArray<std::string,Char8>(fileH, size, EIOD::CHAR, f);
+}
+
+// functions for reading formatted files .FEGRID,  .FINIT,  ...
+// not part of PR 633, hence not reviewed. tskille 2019/02/19
+
+std::vector<std::string> split_string(const std::string& inputStr)
+{
+    std::istringstream iss(inputStr);
+
+    std::vector<std::string> tokens {std::istream_iterator<std::string>{iss},
+                                     std::istream_iterator<std::string>{}
+                                    };
+
+    return tokens;
+}
+
+
+
+void readFormattedHeader(std::fstream& fileH, std::string& arrName,
+                         int &num, EIOD::eclArrType &arrType)
+{
+    if (!fileH.is_open()) {
+        OPM_THROW(std::runtime_error, "fstream fileH not open for reading");
+    }
+
+    std::string line;
+    getline(fileH,line);
+
+    int p1 = line.find_first_of("'");
+    int p2 = line.find_first_of("'",p1+1);
+    int p3 = line.find_first_of("'",p2+1);
+    int p4 = line.find_first_of("'",p3+1);
+
+    if (p1 == -1 || p2 == -1 || p3 == -1 || p4 == -1) {
+        OPM_THROW(std::runtime_error, "Header name and type should be enclosed with '");
+    }
+
+    arrName = line.substr(p1 + 1, p2 - p1 - 1);
+    std::string antStr = line.substr(p2 + 1, p3 - p2 - 1);
+    std::string arrTypeStr = line.substr(p3 + 1, p4 - p3 - 1);
+
+    num = std::stoi(antStr);
+
+    if (arrTypeStr == "INTE")
+        arrType = EIOD::INTE;
+    else if (arrTypeStr == "REAL")
+        arrType = EIOD::REAL;
+    else if (arrTypeStr == "DOUB")
+        arrType = EIOD::DOUB;
+    else if (arrTypeStr == "CHAR")
+        arrType = EIOD::CHAR;
+    else if (arrTypeStr == "LOGI")
+        arrType = EIOD::LOGI;
+    else if (arrTypeStr == "MESS")
+        arrType = EIOD::MESS;
+    else
+        OPM_THROW(std::runtime_error, "Error, unknown array type '" + arrTypeStr +"'");
+
+    if (arrName.size() != 8) {
+        OPM_THROW(std::runtime_error, "Header name should be 8 characters");
+    }
+}
+
+
+template<typename T>
+std::vector<T> readFormattedArray(std::fstream& fileH, const int size,
+                                 std::function<T(const std::string&)>& process)
+{
+    std::vector<T> arr;
+    std::vector<std::string> tokens;
+    std::string line;
+
+    if (!fileH.is_open()) {
+        OPM_THROW(std::runtime_error, "fstream fileH not open for reading");
+    }
+
+    int num = 0;
+    while (num < size) {
+        if (fileH.eof()) {
+            OPM_THROW(std::runtime_error, "End of file reached when reading array");
         }
 
-        fileH.read((char *)&dtail, sizeof(dtail));
-        dtail=flipEndianInt(dtail);
- 
-        if (dhead!=dtail){ 
-            std::string message="Error reading binary doub data, tail not matching header.";
-            throw std::runtime_error(message);
+        getline(fileH, line);
+        tokens = split_string(line);
```

Comment:
I don't see the `split_string` implementation here - but does it handle embedded strings; i.e. this is a typical formatted data from a `.SMSPEC` file:
```
'W1      ' 'W2      ' '+-+-+-+-' 
```